src/common/components/elements/ListingPreview.tsx

```diff
@@ -0,0 +1,364 @@
+
+  if (timeDiffMs < 0) return <span></span>;
+  const lessThanADay = timeDiffMs < 86400000; // one day in ms
+  if (lessThanADay) {
+    // only return the "expensive" Countdown component if required
+    return <Countdown endTime={props.endTime} />;
+  } else {
+    const timeLeft = calculateTimeLeft(props.endTime).toFormat('dd:hh:mm:ss');
+
+    const daysLeft2 = Number(timeLeft.slice(0, 2));
+
+    return (
+      <span>
+        Ends in {daysLeft2} day{daysLeft2 > 1 && 's'}
+      </span>
+    );
+  }
+}
+
+// adds the active loading animation to the antd skeleton image
+const StyledSkeletonImage = styled(Skeleton.Image)`
+  background: linear-gradient(
+    90deg,
+    rgba(34, 34, 34, 0.2) 25%,
+    rgba(255, 255, 255, 0.16) 37%,
+    rgba(34, 34, 34, 0.2) 63%
+  );
+  background-size: 400% 100%;
+  animation: ant-skeleton-loading 1.4s ease infinite;
+  border-radius: 8px;
+`;
+
+export function generateListingShell(id: string): Listing {
+  const now = new Date().toISOString();
+  const nextWeek = new Date(now).toISOString();
+
+  return {
+    address: id,
+    ended: false,
+    items: [
+      {
+        address: '',
+        name: '',
+        uri: '',
+      },
+    ],
+    created_at: now,
+    ends_at: nextWeek,
+    subdomain: '',
+    storeTitle: '',
+  };
+}
+
+// Going with a full replace of the listing during loading for now, but might revert to swapping individual parts of the component below with its loading state. (as done in a previous commit)
+export function SkeletonListing() {
+  return (
+    <ListingPreviewContainer>
+      <Square>
+        <StyledSkeletonImage style={{ borderRadius: '8px', width: '100%', height: '100%' }} />
+      </Square>
+      <Row justify="space-between">
+        <Skeleton.Button size={'small'} active />
+        <Skeleton.Button size={'small'} active />
+      </Row>
+      <Row justify="space-between">
+        <Skeleton.Button size={'small'} active />
+        <Skeleton.Button size={'small'} active />
+      </Row>
+    </ListingPreviewContainer>
+  );
+}
+
+const CustomImageMask = styled.div`
+  position: absolute;
+  right: 0;
+  bottom: 0;
+  width: 25%;
+  height: 25%;
+  margin: auto;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+`;
+
+const captureCid = /https:\/\/(.*).ipfs.dweb.*$/;
+const maybeCDN = (uri: string) => {
+  const cdnURI = uri.replace(captureCid, `${process.env.NEXT_PUBLIC_IPFS_CDN_HOST}/$1`);
+  return cdnURI ?? uri;
+};
+
+export function ListingPreview(listing: Listing) {
+  const storeHref = `https://${listing?.subdomain}.holaplex.com/#/auction/${listing.address}`;
+
+  const [showArtPreview, setShowArtPreview] = useState(false);
+  const [nft, setNFT] = useState<NFTMetadata | null>(null);
+
+  const nftMetadata = listing?.items?.[0]; // other items are usually tiered auctions or participation nfts
+  useEffect(() => {
+    async function fetchNFTDataFromIPFS() {
```

Comment:
We should work to get this as a method on the metadata class of the metaplex js sdk.